### PR TITLE
Allow user without mic or cam to join conference

### PR DESCRIPTION
**Pull request by niuyunsi**

### CODE REVIEW

File: src/components/Conference.tsx [hunk 1]
```diff
@@ -89,9 +89,9 @@ export interface IConferenceProps {
     onError?: (error: ConferenceError) => void;
 }
 
-const webcamScreenConstraints = {
-    audio: true,
-    video: true,
+const sdpConstraints = {
```
Comment (linweigao) [inline src/components/Conference.tsx]: SDPConstraints

File: src/components/Conference.tsx [hunk 2]
```diff
@@ -596,16 +606,23 @@ export class Conference extends React.Component<IConferenceProps, IConferenceSta
         }
         const peerConnection = this.createPeerConnectionById(id);
 
+        // NOTE(yunsi): Initialize remote stream
+        this.createRemoteStreamById(id);
```
Comment (linweigao) [inline src/components/Conference.tsx]: Why?

File: src/components/Conference.tsx [hunk 3]
```diff
@@ -429,16 +429,26 @@ export class Conference extends React.Component<IConferenceProps, IConferenceSta
...
+                navigator.mediaDevices.getUserMedia(constrains)
+                    .then(stream => {
+                        this.localCamStream = stream;
+                        this.setLocalStream(stream, {
+                            isScreenSharing: false
+                        });
+                    })
+                    .catch(this.handleMediaException);
+            } else {
+                this.connection.subscribe(this.handleIncomingMessage);
```
Comment (linweigao) [inline src/components/Conference.tsx]: Is this for subscribing websocket message?

File: src/components/Conference.tsx [hunk 4]
```diff
@@ -695,11 +712,7 @@ export class Conference extends React.Component<IConferenceProps, IConferenceSta
                     ...this.state.remoteStreams,
                     [id]: {
                         ...this.state.remoteStreams[id],
-                        id: id,
                         stream: event.stream,
-                        local: false,
```
Comment (linweigao) [inline src/components/Conference.tsx:700]: Will this be an issue?

File: src/components/Conference.tsx (hunk 2 above)
Comment (niuyunsi) [reply src/components/Conference.tsx]: Change reverted

File: src/components/Conference.tsx (hunk 1 above)
Comment (niuyunsi) [reply src/components/Conference.tsx]: Done

**Review by linweigao — APPROVED** [review]

**Pull request merged**